src/init.luau

```diff
@@ -165,13 +189,20 @@ local function decode(refId: number, encoded: string, state: DecodeState)
 	elseif valueType == "Vector3" then
 		return Vector3.new(string.unpack("<nnn", value))
 	elseif valueType == "CFrame" then
-		return CFrame.new(string.unpack(`<{string.rep("nnn", 4)}`, value))
+		local cframeCase, pointer = string.unpack("<B", value)
+		if cframeCase ~= 0 then
+			local x, y, z = string.unpack("<nnn", value, pointer)
+			return CFrame.new(x, y, z) * CFRAME_SPECIAL_CASES[cframeCase]
+		else
+			local x, y, z, R00, R01, R02, R10, R11, R12, R20, R21, R22 = string.unpack(`<{string.rep("nnn", 4)}`, value, pointer)
```

Comment:
Maybe consider using `CFrame.fromMatrix` here as well, and loading `XVector`, `YVector`, and `ZVector` separately instead? I think this would be cleaner and more readable. I believe this would require serializing position, `XVector`, `YVector`, and `ZVector` above since I don't think the components are in the same order (pretty sure R00 R10 R20 makes up XVector instead of R00 R01 R02 but I might be wrong, should definitely check this)

Reply:
I'm not too sure how I would achieve this whilst retaining the same performance, as if i'm not wrong, you want construct 4 `Vector3`s (whilst unpacking 4 separate times?)


According to the [CFrame docs](https://create.roblox.com/docs/reference/engine/datatypes/CFrame), R00, R10 and R20 make up to RightVector, but the XVector is made up of R00, R01 and R02.
![image](https://user-images.githubusercontent.com/132710510/236703626-2c9f9233-9739-4ba8-a73d-a833cafe48f7.png)